plugin/restore/mysql/mysql.go

```diff
@@ -520,9 +520,45 @@ func binlogFilesAreContinuous(files []BinlogFile) bool {
 	return true
 }
 
-// FetchArchivedBinlogFiles downloads the missing binlog files from the remote instance to `binlogDir`,
-// but exclude latest binlog. We may  download the latest binlog only when doing PITR.
-func (r *Restore) FetchArchivedBinlogFiles(ctx context.Context) error {
+// Download binlog files on server.
+func (r *Restore) downloadBinlogFilesOnServer(ctx context.Context, binlogFilesLocal, binlogFilesOnServerSorted []BinlogFile) error {
+	if len(binlogFilesOnServerSorted) == 0 {
+		return fmt.Errorf("no binlog files on server")
```

Comment:
FetchArchivedBinlogFiles() doesn't fail when there's no binlog files on the server. It's possible that binary logs are all purged but cached locally. We should not fail in this case.

Reply:
Fixed.